site/src/api/queries/deployment.ts

```diff
@@ -0,0 +1,15 @@
+import * as API from "api/api";
+
+export const deploymentConfig = () => {
+  return {
+    queryKey: ["deployment", "config"],
```

Comment:
Could the query keys be declared as const? That would not only make sure that they're kept immutable, but also give more specific type information when accessing elements of the key at specific indices

Reply:
I don't know if this would cause the compiler to flag anything, though. I'd mainly be worried about interfacing readonly arrays with normal arrays

Reply:
I think they could be but I don't see to much value on doing that honestly, what use case you are thinking of?

Reply:
Yeah, I'm looking at the code again, and I think it has limited value here. I think I generally use this approach, but it's most useful when you have a key array that mixes different types.

```ts
const key1 = ["workspaces", 23, { running: true }];
const key2 = ["workspaces", 23, { running: true }] as const;

// Inferred as type (string | number | { running: boolean }), because array is arbitrary-length
const userId1 = key1[1];

// Inferred as type number, because key2 is a fixed-length, immutable tuple
const userId2 = key2[1];
```

Reply:
I see, but I don't see we using something similar like this - accessing the keys directly - so I think we could wait until it becomes an issue or provide more value. Wdyt? But if you feel strong about that, we can do it for all the query keys, I just would do in a diff PR (since it is going to change non related queries as well).